### ISSUE TITLE
factory SpreadsheetDecoder.decodeBuffer

### ISSUE THREAD

**Issue by david-mat** [sestegra/spreadsheet_decoder#36]
InputStream input paramater should be InputStreamBase
This allow calling decodeBuffer with an InputFileStream object

**Comment by sestegra** [comment]
The current implementation of `SpreadsheetDecoder.decodeBuffer` uses `input` as `InputStream` class.
The underlying method used to unzip the content is `ZipDecoder().decodeBuffer` that requires `InputStream` class as well.

``` dart
  factory SpreadsheetDecoder.decodeBuffer(InputStream input,
      {bool update = false, bool verify = false}) {
    var archive = ZipDecoder().decodeBuffer(input, verify: verify);
    return _newSpreadsheetDecoder(archive, update);
  }
```

**Comment by david-mat** [comment]
Hi,

ZipDecoder().decodeBuffer parameter is InputStreamBase. SpreadsheetDecoder.decodeBuffer parameter should be the same.

Please see:
https://api.flutter.dev/flutter/archive_io/ZipDecoder/decodeBuffer.html

**Comment by sestegra** [comment]
The package is [archive](https://pub.dev/documentation/archive/latest/archive_io/ZipDecoder-class.html) and not the one from Flutter SDK.

Nevertheless, its API has a similar change introduced on 3.1.4 release and the current dependency is `archive: ^3.0.0`, so I need to update the `pubspec.yaml` and the implementation.

Good catch and thank you for reporting.

**Comment by david-mat** [comment]
You welcome :-)

**Comment by sestegra** [comment]
Fixed by commit d12d7fba24b50e0f76bd5d25290ab88cfe6aff5c
